### PR TITLE
Add nrfx config files to custom directory

### PR DESCRIPTION
**Pull request by henri-gruender**
Moving the nrfx config files allows a cleaner implementation of configuration changes and prepares for the use of a custom `nrfx_glue.h`.

### CODE REVIEW

File: build_tools/third_party/nrfx/CMakeLists.txt [hunk 1]
```diff
@@ -77,11 +77,12 @@ target_sources(nrfx_uart
 
 target_include_directories(nrfx_uart
   PUBLIC
+    ${nrfx_custom_DIR}
     ${nrfx_DIR}
     ${nrfx_DIR}/drivers/include/
     ${nrfx_DIR}/hal/
-    ${nrfx_DIR}/templates/
     ${nrfx_DIR}/mdk/
+    ${nrfx_DIR}/soc/
```
Comment (marbre) [inline build_tools/third_party/nrfx/CMakeLists.txt:85]: This seems not strictly necessary as files from `${nrfx_custom_DIR}` where not included for this target till now. Anyway, I am generally fine with this change but I would like to have this in a separate commit to allow an easier tracking of changes as well as understanding the motivation behind.

**Review by marbre — CHANGES_REQUESTED** [review]
Thanks for working through this! The changes overall looks good but I would like to see that this patch is split up into multiple commits:

* A commit adding the unmodified, imported files.
* A commit that adjust the files to our needs or rather moves the modifications from `nrf_glue_nrf5340.h` to `nrfx_glue.h`.
* A commit with additional changes, see additional commit.

It would be nice if every commit is able to pass the pipeline, which should be the case. This would allow to rebase and merge instead of doing a squash commit.

**Comment by marbre** [comment]
One additional remark regarding the "import commit", we don't need to care if clang-format fails. This is one of the positive aspects of the additional work ;)

File: build_tools/third_party/nrfx/CMakeLists.txt (hunk 1 above)
Comment (henri-gruender) [reply build_tools/third_party/nrfx/CMakeLists.txt]: Good point. I would rather not include it. It will be an easy change if needed.

**Comment by henri-gruender** [comment]
Thank you for the review @marbre. I hope the splitted commits make sense now. I believe they should all pass the pipeline.

**Review by marbre — APPROVED** [review]
Thanks for splitting this up and testing the commits individually!

**Pull request merged**